src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts

```diff
@@ -385,96 +388,100 @@ export class ServiceFormComponent extends CdForm implements OnInit {
     if (this.editing) {
       this.action = this.actionLabels.EDIT;
       this.disableForEditing(this.serviceType);
-      this.cephServiceService.list(this.serviceName).subscribe((response: CephServiceSpec[]) => {
-        const formKeys = ['service_type', 'service_id', 'unmanaged'];
-        formKeys.forEach((keys) => {
-          this.serviceForm.get(keys).setValue(response[0][keys]);
-        });
-        if (!response[0]['unmanaged']) {
-          const placementKey = Object.keys(response[0]['placement'])[0];
-          let placementValue: string;
-          ['hosts', 'label'].indexOf(placementKey) >= 0
-            ? (placementValue = placementKey)
-            : (placementValue = 'hosts');
-          this.serviceForm.get('placement').setValue(placementValue);
-          this.serviceForm.get('count').setValue(response[0]['placement']['count']);
-          if (response[0]?.placement[placementValue]) {
-            this.serviceForm.get(placementValue).setValue(response[0]?.placement[placementValue]);
-          }
-        }
-        switch (this.serviceType) {
-          case 'iscsi':
-            const specKeys = ['pool', 'api_password', 'api_user', 'trusted_ip_list', 'api_port'];
-            specKeys.forEach((key) => {
-              this.serviceForm.get(key).setValue(response[0].spec[key]);
-            });
-            this.serviceForm.get('ssl').setValue(response[0].spec?.api_secure);
-            if (response[0].spec?.api_secure) {
-              this.serviceForm.get('ssl_cert').setValue(response[0].spec?.ssl_cert);
-              this.serviceForm.get('ssl_key').setValue(response[0].spec?.ssl_key);
+      this.cephServiceService
+        .list(new HttpParams({ fromObject: { limit: -1, offset: 0 } }), this.serviceName)
```

Comment:
@pereman2 do we actually pagination when we call the list endpoint with a known service name?

Reply:
I wasn't properly updating the `httpParams` argument in `cephServiceService.list`. This should be fixed by now, please re review.